From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Tue,  7 Mar 2023 17:46 +0000
Subject: [PATCH] Remove gcc option -no-red-zone for aarch64 builds

gcc on aarch64 does not support -no-red-zone:

   gcc: error: unrecognized command-line option '-mno-red-zone'

So only use this option for x86 and x86_64.

---

diff -aur a/make.conf b/make.conf
--- a/make.conf	2023-03-07
+++ b/make.conf	2023-03-07
@@ -32,10 +32,12 @@
 
 CFLAGS = \
         -Wall -Werror \
-        ${EFIINCS} -fno-stack-protector -fpic -fshort-wchar -mno-red-zone \
+        ${EFIINCS} -fno-stack-protector -fpic -fshort-wchar \
         -DGIT_VERSION=\"$(shell git rev-parse HEAD | cut -c1-12)\"
 ifeq (${ARCH},x86_64)
-CFLAGS += -DEFI_FUNCTION_WRAPPER
+CFLAGS += -mno-red-zone -DEFI_FUNCTION_WRAPPER
+else ifeq (${ARCH},x86)
+CFLAGS += -mno-red-zone
 endif
 LDFLAGS = \
         -shared \
